scipy/optimize/tests/test_trustregion.py

```diff
@@ -102,3 +106,31 @@ def test_trust_exact_start_in_optimum(self):
         r = minimize(rosen, x0=self.x_opt, jac=rosen_der, hess=rosen_hess,
                      tol=1e-8, method='trust-exact')
         assert_allclose(self.x_opt, r['x'])
+
+    def test_finite_differences(self):
+        # if the Hessian is estimated by finite differences or
+        # a HessianUpdateStrategy (and no hessp is provided) then creation
+        # of a hessp is possible.
+        # GH13754
+        methods = ["trust-ncg", "trust-krylov", "dogleg"]
+        product = itertools.product(
+            FD_METHODS + (BFGS,),
+            methods
+        )
+        # a hessian needs to be specified for trustregion
+        for method in methods:
+            with pytest.raises(ValueError):
```

Comment:
Can you match the exception here?

Reply:
It doesn't strike me as essential here, but I'll add something.

Reply:
Have tried that. Doesn't work because the different minimizers have different error messages.

Reply:
Ok no problem then, thanks.